lib/OpenQA/WebSockets/Client.pm

```diff
@@ -61,8 +62,9 @@ sub singleton { state $client ||= __PACKAGE__->new }
 
 sub _api {
     my ($self, $method) = @_;
-    my $port = $self->port;
-    return "http://127.0.0.1:$port/api/$method";
+    my $address = $self->address;
+    my $port    = $self->port;
+    return "http://$address:$port/api/$method";
```

Comment:
Looks good but you need to adjust `OpenQA::Scheduler::Client` in the same way.

Reply:
@okurz